### ISSUE TITLE
Generic.WhiteSpace.ScopeIndent.Incorrect violation identified but fails to fix it

### ISSUE THREAD

**Issue by manderson** [squizlabs/PHP_CodeSniffer#2054]
When phpcs is ran on either of the following files:

```
<?php

if (true) {
class Foo
{
}
}
```

```
<?php

if (false) {
function Foo()
{
}
}
```

The following is reported:
```
5 | ERROR | [x] Line indented incorrectly; expected at least 4 spaces, found 0 (Generic.WhiteSpace.ScopeIndent.Incorrect)

PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
```

But when phpcbf is ran it reports 
```
FAILED TO FIX
```
I'm using 3.3.0 (stable).

**Comment by gsherwood** [comment]
This is most likely a conflict between the ScopeIndent sniff and another sniff. If you try running the ScopeIndent sniff alone, it should work fine:
```
$ cat temp.php
<?php

if (true) {
class Foo
{
}
}
$ phpcs --standard=Generic --sniffs=Generic.WhiteSpace.ScopeIndent temp.php

FILE: temp.php
--------------------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
--------------------------------------------------------------------------------
 4 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 0
 5 | ERROR | [x] Line indented incorrectly; expected at least 4 spaces, found 0
 6 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 0
--------------------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------

Time: 53ms; Memory: 4Mb

$ phpcbf --standard=Generic --sniffs=Generic.WhiteSpace.ScopeIndent temp.php

PHPCBF RESULT SUMMARY
-----------------------------------------------------------------------------
FILE                                                         FIXED  REMAINING
-----------------------------------------------------------------------------
temp.php                                                     3      0
-----------------------------------------------------------------------------
A TOTAL OF 3 ERRORS WERE FIXED IN 1 FILE
-----------------------------------------------------------------------------

Time: 35ms; Memory: 4Mb


$ cat temp.php
<?php

if (true) {
    class Foo
    {
    }
}
```
The easiest way to debug this is to run phpcs over the sample file and use `--report=diff -vvv`. This will output a lot of content, but it will also show you (at the end) which sniffs are conflicting and what changes they are making. If you can post the last few loops of file fixing, that should tell me what is going on.

**Comment by manderson** [comment]
Attached is an output file generated from the command:

```
phpcs --report=diff -vvv --standard=~/config/dsr_ruleset.phpcs.xml foo.php > output.txt
```

[output.txt](https://github.com/squizlabs/PHP_CodeSniffer/files/2086690/output.txt)

**Comment by gsherwood** [comment]
Thanks for generating that output. The problem comes from the fact that you are excluding the `Generic.WhiteSpace.ScopeIndent.IncorrectExact` error code, which stops the fixer indenting the class. But you letting the other `Generic.WhiteSpace.ScopeIndent.Incorrect` error code through, so the closing brace is indented, which then sets off the conflict.

Here is what fixing looks like when that error message isn't excluded:
```
	*** START FILE FIXING ***
---START FILE CONTENT---
1|<?php
2|
3|if (false) {
4|class Foo
5|{
6|}
7|}
8|
--- END FILE CONTENT ---
	E: [Line 4] Line indented incorrectly; expected 4 spaces, found 0 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
	PHP_CodeSniffer\Standards\Generic\Sniffs\WhiteSpace\ScopeIndentSniff (line 1342) replaced token 10 (T_CLASS) "class" => "····class"
	E: [Line 5] Line indented incorrectly; expected at least 4 spaces, found 0 (Generic.WhiteSpace.ScopeIndent.Incorrect)
	PHP_CodeSniffer\Standards\Generic\Sniffs\WhiteSpace\ScopeIndentSniff (line 1342) replaced token 14 (T_OPEN_CURLY_BRACKET) "{" => "····{"
	E: [Line 6] Line indented incorrectly; expected 4 spaces, found 0 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
	PHP_CodeSniffer\Standards\Generic\Sniffs\WhiteSpace\ScopeIndentSniff (line 1342) replaced token 16 (T_CLOSE_CURLY_BRACKET) "}" => "····}"
	* fixed 3 violations, starting loop 2 *
---START FILE CONTENT---
1|<?php
2|
3|if (false) {
4|    class Foo
5|    {
6|    }
7|}
8|
--- END FILE CONTENT ---
	*** END FILE FIXING ***
--- temp.php
+++ PHP_CodeSniffer
@@ -1,7 +1,7 @@
 <?php

 if (false) {
-class Foo
-{
-}
+    class Foo
+    {
+    }
 }
```

I'm not sure why you are excluding that error code as it's not one people normally exclude, but auto-fixing wont work without it, so you'll need to add it back in to get this working.

If you've got a problem with that code, please let me know and I'll see if I can help.
